src/o365/spfx/commands/project/project-rename.ts

```diff
@@ -0,0 +1,247 @@
+      {
+        option: '--generateNewId',
+        description: 'Generate a new solution id for the project'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (!args.options.newName) {
+        return 'Required parameter newName missing';
+      }
+
+      return true;
+    };
+  }
+
+  public action(): CommandAction {
+    const cmd: Command = this;
+    return function (this: CommandInstance, args: CommandArgs, cb: (err?: any) => void) {
+      args = (cmd as any).processArgs(args);
+      (cmd as any).initAction(args, this);
+      cmd.commandAction(this, args, cb);
+    }
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      cb(new CommandError(`Couldn't find project root folder`, SpfxProjectRenameCommand.ERROR_NO_PROJECT_ROOT_FOLDER));
+      return;
+    }
+
+    const packageJson: any = this.getProject(this.projectRootPath).packageJson;
+    const projectName: string = packageJson.name;
+
+    let newId: string = '';
+    if (args.options.generateNewId) {
+      newId = this.generateNewId();
+      if (this.debug) {
+        cmd.log('Created new solution id');
+        cmd.log(newId);
+      }
+    }
+
+    if (this.debug) {
+      cmd.log(`Renaming SharePoint Framework project to '${args.options.newName}'`);
+    }
+
+    try {
+      this.replacePackageJsonContent(path.join(this.projectRootPath, 'package.json'), args, cmd);
+      this.replaceYoRcJsonContent(path.join(this.projectRootPath, '.yo-rc.json'), newId, args, cmd);
+      this.replacePackageSolutionJsonContent(path.join(this.projectRootPath, 'config', 'package-solution.json'), projectName, newId, args, cmd);
+      this.replaceDeployAzureStorageJsonContent(path.join(this.projectRootPath, 'config', 'deploy-azure-storage.json'), args, cmd);
+      this.replaceReadMeContent(path.join(this.projectRootPath, 'README.md'), projectName, args, cmd);
+    } catch (error) {
+      cb(new CommandError(error));
+      return;
+    }
+
+    if (this.verbose) {
+      cmd.log('DONE');
+    }
+
+    cb();
+  }
+
+  private generateNewId = (): string => {
+    return v4();
+  }
+
+  private replacePackageJsonContent = (filePath: string, args: CommandArgs, cmd: CommandInstance) => {
+    if (!fs.existsSync(filePath)) {
+      return;
+    }
+    const existingContent = fs.readFileSync(filePath, 'utf-8');
+    const updatedContent = JSON.parse(existingContent);
+    if (updatedContent &&
+      updatedContent.name) {
+      updatedContent.name = args.options.newName;
+    }
+    if (JSON.stringify(updatedContent, null, 2) !== JSON.stringify(existingContent, null, 2)) {
+      fs.writeFileSync(filePath, JSON.stringify(updatedContent, null, 2), 'utf-8');
```

Comment:
Since we stringified the object previously, we should reuse it rather than stringifying it again

Reply:
Agreed. Updated now.